app/vmselect/promql/rollup.go

```diff
@@ -900,17 +900,26 @@ func getMaxPrevInterval(scrapeInterval int64) int64 {
 	return scrapeInterval + scrapeInterval/8
 }
 
-func removeCounterResets(values []float64) {
+func removeCounterResets(values []float64, timestamps []int64, maxStalenessInterval int64) {
 	// There is no need in handling NaNs here, since they are impossible
 	// on values from vmstorage.
 	if len(values) == 0 {
 		return
 	}
 	var correction float64
 	prevValue := values[0]
+	prevTS := timestamps[0]
 	for i, v := range values {
 		d := v - prevValue
 		if d < 0 {
+			currTS := timestamps[i]
+			if maxStalenessInterval > 0 && ((currTS - prevTS) > maxStalenessInterval) {
```

Comment:
As I understood, this check must be performed also if `d>=0`. For example, if the duration between the previous and the current sample exceeds `maxStalenessInterval` and the previous sample value is 10 and the current sample value is 11, then the current sample value must be set to 11, while the `correction` must be set to 0.

Reply:
Thanks! That should be addressed in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/8073/commits/4a590626907cc4a0e8bed35912cefa4e9440b0f3